src/object.cr

```diff
@@ -201,6 +201,23 @@ class Object
     yield self
   end
 
+  # Returns `true` if `self` is included in the *collection* argument.
+  #
+  # ```
+  # 10.in?(0..100)     # => true
+  # 10.in?({0, 1, 10}) # => true
+  # 10.in?(0, 1, 10)   # => true
+  # 10.in?(:foo, :bar) # => false
+  # ```
+  def in?(collection : Enumerable) : Bool
+    collection.includes?(self)
+  end
+
+  # :ditto:
+  def in?(*values : Object) : Bool
```

Comment:
I'd rather drop this type restriction too

Reply:
That's on purpose - otherwise it will be possible to call it with no arguments.

Reply:
Hm, that's weird. Why does having a type restriction enforce size > 0...

Alternatively, this should work: `def in?(value, *values)`. It's not as nice, though.

Reply:
> Hm, that's weird. Why does having a type restriction enforce size > 0...

I dunno, it's probably better to ask @asterite about that.

>Alternatively, this should work: `def in?(value, *values)`. It's not as nice, though.

It's not. I'd rather leave it like it is right now, regardless the perceived _weirdness_.

Reply:
Because if you specify a type, any empty tuple would match it, and if you have multiple overloads with different type restrictions it's not clear which should match. So the solution is: none match. The empty tuple doesn't match a splat with a type restriction. And if you want to match that you provide another overload without args.

And specifying `Object` is a convenient way of saying "match one or more args".

Reply:
I think this overload is confusing and I'd remove it.

```crysta
1.in?([1, 2, 3], 4) # => false
1.in?([1, 2, 3]) # => true
```

That is, it's not clear that the one-argument version is a specialization of the multi-args but with one arg, or a completely different behavior.

You can always do `elem.in?({a, b, c})`

Reply:
@asterite wow! Thanks I never knew this trick.

Reply:
@asterite IMO this method will be most be used in place of `{:foo, :bar, :baz}.includes?(val)` notation (used often with `Errno#errno` or `HTTP::Client::Response#status_code`), mixing the splats and arguments is sth more of an edge-case.